src/bootstrap/src/core/build_steps/run.rs

```diff
@@ -154,7 +154,7 @@ impl Step for Miri {
         miri.args(builder.config.args());
 
         // miri tests need to know about the stage sysroot
-        miri.env("MIRI_SYSROOT", &miri_sysroot);
+        miri.arg("--sysroot").arg(miri_sysroot);
```

Comment:
This should go before the `builder.config.args()`, so that the user can do something like `./x.py run miri -- flags-for-miri -- flags-for-the-program`. Your approach adds the `--sysroot`  at the end, but due to the `--` that doesn't work properly.

Reply:
right. i cannot be trusted with passing arguments.